src/pages/AdvancedSearch/SearchForm/SearchHistoryMenu/SearchHistoryMenu.tsx

```diff
@@ -14,26 +13,26 @@ export enum Label {
 
 type Props = {
   queryHistory: string[];
+  search: (query: string) => void;
   setQueryHistory: (queryHistory: string[]) => void;
 };
 
 const SearchHistoryMenu = ({
   queryHistory,
+  search,
   setQueryHistory,
 }: Props): JSX.Element => {
   const loading = useAppSelector(getCrossModelQueryLoading);
   const { setFieldValue } = useFormikContext<FormFields>();
-  const [, setQueryParams] = useQueryParams<{ q: string }>({
-    q: "",
-  });
 
   return (
     <ContextualMenu
       links={[
         ...queryHistory.map((query) => ({
           children: query,
+          disabled: loading,
```

Comment:
I changed this component so that it disables the menu items rather than the button that opens the menu as it didn't quite feel right how it was before.

Reply:
Just tested and the user experience definitely feels better with this change!